.gitignore

```diff
@@ -1,9 +1,57 @@
+*.iml
+*.idea/**/*
+*.idea
+.idea/*
```

Comment:
`.idea` will let us git-ignore all the files under `.idea`, what is this complex setup for?